split-llvm-extract.cpp

```diff
@@ -247,6 +247,10 @@ int main(int argc, char **argv)
 	{
 		if (function.isDeclaration())
 		{
+			if (function.getVisibility() != llvm::GlobalValue::VisibilityTypes::ProtectedVisibility)
```

Comment:
Given there only seem to be 3 visibility types (https://llvm.org/doxygen/classllvm_1_1GlobalValue.html#a9141f967188383108a69cc1b8ed3c195) this is really saying, I think, `if (function.getVisibility() == llvm::GlobalValue::VisibilityTypes::HiddenVisibility)`? That seems a very big stick with which to hit things, and seems like it might make things visible that shouldn't be (which decreases security).

Can you explain more precisely if/why this is safe/correct?